src/core/service.c

```diff
@@ -1248,15 +1249,6 @@ static void service_set_state(Service *s, ServiceState state) {
                 unit_dequeue_rewatch_pids(UNIT(s));
         }
 
-        if (!IN_SET(state,
-                    SERVICE_CONDITION, SERVICE_START_PRE, SERVICE_START, SERVICE_START_POST,
-                    SERVICE_RUNNING,
-                    SERVICE_RELOAD, SERVICE_RELOAD_SIGNAL, SERVICE_RELOAD_NOTIFY,
-                    SERVICE_STOP, SERVICE_STOP_WATCHDOG, SERVICE_STOP_SIGTERM, SERVICE_STOP_SIGKILL, SERVICE_STOP_POST,
-                    SERVICE_FINAL_WATCHDOG, SERVICE_FINAL_SIGTERM, SERVICE_FINAL_SIGKILL) &&
-            !(state == SERVICE_DEAD && UNIT(s)->job))
```

Comment:
Did you mean to reply below? Anyway, I was thinking of a "global" cap, would that be useful?

Reply:
i have no idea why github inserted my reply here. this is frciking weird

Reply:
> these checks seem different from what is now done in service_close_socket_fd() and service_release_resources(), is that intentional?

That's because service_release_resources() is called from unit_releas_resources() which already does similar checks, so we don't need to list most states there, nor check for the jobs.